sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/FunctionToolCallingSample.java

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai;
+
+import com.azure.ai.openai.models.ChatChoice;
+import com.azure.ai.openai.models.ChatCompletions;
+import com.azure.ai.openai.models.ChatCompletionsFunctionToolCall;
+import com.azure.ai.openai.models.ChatCompletionsFunctionToolDefinition;
+import com.azure.ai.openai.models.ChatCompletionsOptions;
+import com.azure.ai.openai.models.ChatCompletionsToolCall;
+import com.azure.ai.openai.models.ChatRequestAssistantMessage;
+import com.azure.ai.openai.models.ChatRequestMessage;
+import com.azure.ai.openai.models.ChatRequestSystemMessage;
+import com.azure.ai.openai.models.ChatRequestUserMessage;
+import com.azure.ai.openai.models.ChatResponseMessage;
+import com.azure.ai.openai.models.FunctionCall;
+import com.azure.ai.openai.models.FunctionDefinition;
+import com.azure.ai.openai.models.FunctionParameters;
+import com.azure.ai.openai.models.FunctionProperties;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.util.BinaryData;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Sample demonstrates how to do a single and parallel function calling via 'tools' calling.
+ */
+public class FunctionToolCallingSample {
+
+    /**
+     * Runs the sample algorithm and demonstrates how to get chat completions using function tool calling.
+     *
+     * @param args Unused. Arguments to the program.
+     */
+    public static void main(String[] args) {
+        String endpoint = "{azure-open-ai-endpoint}";
+        String azureOpenaiKey = "{azure-open-ai-key}";
+        // Newer models like gpt-4-1106-preview or gpt-3.5-turbo-1106 can call multiple functions in one turn.
+        String deploymentOrModelId = "{azure-open-ai-deployment-model-id}";
+
+        // Function call is supported for model versions with suffix `-0613` and later.
+        OpenAIClient client = new OpenAIClientBuilder()
+                .endpoint(endpoint)
+                .credential(new AzureKeyCredential(azureOpenaiKey))
+                .buildClient();
+
+        singleFunctionCallingSample(client, deploymentOrModelId);
+        parallelFunctionCallingSample(client, deploymentOrModelId);
```

Comment:
I think this should be separated into 2 samples. It's really hard to follow what you are trying to focus on here. I would suggest splitting this into `ChatCompletionsSingleToolCall` and `ChatCompletionsParallelToolCall`. 

Here you seem to be conflating `function_call` with `tool_call`. Depending how you setup your request, these will produce different `finishe_reasons` , AFAIK.

Reply:
I would delete the single case since it is easy to convert to single from multiple